Errors with version 4.0 and material-ui 4

Installing version 4.0 @rjsf/core and @rjsf/material-ui, alongside with my material-ui 4, I get the warning

`npm WARN @rjsf/material-ui@4.0.0 requires a peer of @rjsf/core@^3.0.0 but none is installed. You must install peer dependencies yourself.`

(and indeed, in the package.json of @rjsf/material-ui one finds "@rjsf/core@^3.0.0" , that seems wrong)

Furthermore, if I ignore the install warning , npm start gives compile warnings

```
./node_modules/@rjsf/material-ui/dist/material-ui.esm.js
Module not found: Can't resolve '@mui/icons-material' in 'C:\wk\pcnt\trake\trake-ui\node_modules\@rjsf\material-ui\dist'

./node_modules/@rjsf/material-ui/dist/material-ui.esm.js
Module not found: Can't resolve '@mui/material' in 'C:\wk\pcnt\trake\trake-ui\node_modules\@rjsf\material-ui\dist'   
```

... and when runnning I get on console

```
Error: Cannot find module '@mui/material'
    at webpackMissingModule (material-ui.esm.js:1256:1)
    at Module../node_modules/@rjsf/material-ui/dist/material-ui.esm.js (material-ui.esm.js:1256:1)
```

Using react 17 and Node 14 with

@material-ui/core: 4.12.3
@material-ui/system: 4.12.1
@material-ui/icons 4.11.2
@rjsf/core: 4.0.0
@rjsf/material-ui: 4.0.0

@epicfaace In my own build I am also seeing these which I believe my fix eliminates as well:
```
      ModuleNotFoundError: Cannot find module '@mui/material' from 'node_modules/@rjsf/material-ui/dist/material-ui.cjs.development.js'
          at Resolver.resolveModule (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-resolve/build/resolver.js:324:11)
          at Resolver._getVirtualMockPath (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-resolve/build/resolver.js:471:14)
          at Resolver._getAbsolutePath (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-resolve/build/resolver.js:457:14)
          at Resolver.getModuleID (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-resolve/build/resolver.js:424:31)
          at Runtime._shouldMock (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-runtime/build/index.js:1917:37)
          at Runtime.require [as requireModuleOrMock] (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-runtime/build/index.js:1204:14)
          at Object.<anonymous> (/home/travis/build/freenome/freenome-core-ui/node_modules/@rjsf/material-ui/src/Theme5/Mui5Context.tsx:10:9)
          at Runtime._execModule (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-runtime/build/index.js:1646:24)
          at Runtime._loadModule (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-runtime/build/index.js:1185:12)
          at Runtime.requireModule (/home/travis/build/freenome/freenome-core-ui/node_modules/jest-runtime/build/index.js:1009:12) {
        code: 'MODULE_NOT_FOUND',
        hint: '',
        requireStack: undefined,
        siblingWithSimilarExtensionFound: false,
        moduleName: '@mui/material',
        _originalMessage: "Cannot find module '@mui/material' from 'node_modules/@rjsf/material-ui/dist/material-ui.cjs.development.js'"
      }
      at Object.<anonymous> (node_modules/@rjsf/material-ui/src/Theme5/Mui5Context.tsx:13:11)
```

I'm still getting `Module not found: Can't resolve '@material-ui/core'` warnings when using 4.0.1. 

```ts
import { withTheme } from '@rjsf/core';
import { Theme5 as Mui5Theme } from '@rjsf/material-ui';

export const Form = withTheme(Mui5Theme);
```

--- a/packages/material-ui/src/Theme/MaterialUIContext.tsx
+++ b/packages/material-ui/src/Theme/MaterialUIContext.tsx
@@ -1,5 +1,5 @@
 import React from 'react';
-import MaterialUIContextProps from 'Theme/MaterialUIContextProps';
+import MaterialUIContextProps from './MaterialUIContextProps';
 
 /** Use require for loading these libraries in case they are not available in order to perform a useful fallback
  */
@@ -9,9 +9,7 @@ try {
   mui = require('@material-ui/core');
   icons = require('@material-ui/icons');
 } catch (err) {
-  if (process.env.NODE_ENV !== 'PRODUCTION') {
-    console.log(err);
-  }
+  // purposely a no-op. If it is not here, don't make noise like we used to
 }
 
 export let MaterialUIContext: MaterialUIContextProps;

--- a/packages/material-ui/src/Theme/MaterialUIContextProps.ts
+++ b/packages/material-ui/src/Theme/MaterialUIContextProps.ts
@@ -1,4 +1,14 @@
-import {
+/** Use require for loading these libraries in case they are not available in order to perform a useful fallback
+ */
+let mui;
+try {
+  mui = require('@material-ui/core');
+} catch (err) {
+  // purposely a no-op
+}
+
+// @ts-ignore What we are doing here isn't really good Typescript, but it works
+const {
   Box,
   Button,
   Checkbox,
@@ -24,11 +34,11 @@ import {
   SvgIcon,
   TextField,
   Typography,
-} from '@material-ui/core';
+} = mui;
 
 export default interface MaterialUIContextProps {
-  Button?: Button;
   Box?: Box;
+  Button?: Button;
   Checkbox?: Checkbox;
   Divider?: Divider;
   Grid?: Grid;

--- a/packages/material-ui/src/Theme/MaterialUIContextProps.ts
+++ b/packages/material-ui/src/Theme/MaterialUIContextProps.ts
@@ -1,6 +1,6 @@
 /** Use require for loading these libraries in case they are not available in order to perform a useful fallback
  */
-let mui;
+let mui = {};
 try {
   mui = require('@material-ui/core');
 } catch (err) {

--- a/packages/material-ui/src/Theme5/Mui5Context.tsx
+++ b/packages/material-ui/src/Theme5/Mui5Context.tsx
@@ -10,9 +10,7 @@ try {
   mui = require('@mui/material');
   icons = require('@mui/icons-material');
 } catch (err) {
-  if (process.env.NODE_ENV !== 'PRODUCTION') {
-    console.log(err);
-  }
+  // purposely a no-op
 }
 
 export let Mui5Context: Mui5ContextProps;

--- a/packages/material-ui/src/Theme5/Mui5ContextProps.ts
+++ b/packages/material-ui/src/Theme5/Mui5ContextProps.ts
@@ -1,6 +1,6 @@
 /** Use require for loading these libraries in case they are not available in order to perform a useful fallback
  */
-let mui;
+let mui = {};
 try {
   mui = require('@mui/material');
 } catch (err) {

--- a/packages/material-ui/src/Theme5/Mui5ContextProps.ts
+++ b/packages/material-ui/src/Theme5/Mui5ContextProps.ts
@@ -1,4 +1,14 @@
-import {
+/** Use require for loading these libraries in case they are not available in order to perform a useful fallback
+ */
+let mui;
+try {
+  mui = require('@mui/material');
+} catch (err) {
+  // purposely a no-op
+}
+
+// @ts-ignore What we are doing here isn't really good Typescript, but it works
+const {
   Box,
   Button,
   Checkbox,
@@ -24,11 +34,11 @@ import {
   SvgIcon,
   TextField,
   Typography,
-} from '@mui/material';
+} = mui;
 
 export default interface Mui5ContextProps {
-  Button?: Button;
   Box?: Box;
+  Button?: Button;
   Checkbox?: Checkbox;
   Divider?: Divider;
   Grid?: Grid;

--- a/packages/material-ui/src/index.ts
+++ b/packages/material-ui/src/index.ts
@@ -12,5 +12,6 @@ export { default as Widgets } from './Widgets';
 
 export { default as Theme5 } from './Theme5';
 export { default as MuiForm5 } from './MuiForm5';
+export { default as MuiComponentContext } from './MuiComponentContext';
 
 export default MuiForm;

